pkg/provider/acme/provider.go

```diff
@@ -468,16 +548,69 @@ func (p *Provider) watchNewDomains(ctx context.Context) {
 	})
 }
 
-func (p *Provider) resolveCertificate(ctx context.Context, domain types.Domain, tlsStore string) (*certificate.Resource, error) {
-	domains, err := p.getValidDomains(ctx, domain)
+func (p *Provider) resolveDefaultCertificate(ctx context.Context, domains []string) (*certificate.Resource, error) {
+	logger := log.FromContext(ctx)
+
+	p.resolvingDomainsMutex.Lock()
+
+	sort.Strings(domains)
+	domainKey := strings.Join(domains, ",")
+
+	if _, ok := p.resolvingDomains[domainKey]; ok {
+		p.resolvingDomainsMutex.Unlock()
+		return nil, nil
+	}
+
+	p.resolvingDomains[domainKey] = struct{}{}
+
+	for _, certDomain := range domains {
+		p.resolvingDomains[certDomain] = struct{}{}
+	}
+
+	p.resolvingDomainsMutex.Unlock()
```

Comment:
Would it maybe be better to refactor this locked segment into a separate function that could `defer` unlocking?

Reply:
technical refactor are a bad practice.

Reply:
One could argue a single +754/-240 line commit with the message "feat: acme default cert" is also bad practice...

Reply:
One could say that I was in a voice chat with the other maintainers and I converted the vocal reviews into commits...

Reply:
I kind of do know the background, being the author of #8123 and #8124, which are referred to in the PR message. :)

My point is it'll be quite difficult for a future maintainer or contributor to understand why there's been ~700 lines of changes if there's just a single-line commit message.

Reply:
I think you didn't understand what I said.
It does not matter, the feature is now inside Traefik, this is the most important thing.